tools/xmlschema.py

```diff
@@ -0,0 +1,312 @@
+
+    lines.append("</xsd:element>")
+    lines.append("</xsd:choice>")
+    return lines
+
+
+def print_attribute(element: ElementTree.Element) -> List[str]:
+    """
+    Print an attribute of the sdf definition
+    """
+    lines = []
+
+    elem_name = get_attribute(element, "name")
+    elem_type = get_attribute(element, "type")
+    elem_reqd = get_attribute(element, "required")
+    elem_default = get_attribute(element, "default")
+
+    if elem_type and is_std_type(elem_type):
+        elem_type = xsd_type_string(elem_type)
+
+    use = ""
+    default = ""
+
+    if elem_reqd == "1":
+        use = "use='required'"
+    elif elem_reqd == "0":
+        use = "use='optional'"
+        if elem_default is not None:
+            default = f"default='{elem_default}'"
+
+    lines.append(
+        f"<xsd:attribute name='{elem_name}' type='{elem_type}' {use} {default}>"
+    )
+    lines.extend(indent_lines(print_documentation(element), 2))
+    lines.append("</xsd:attribute>")
+    return lines
+
+
+def print_xsd(element: ElementTree.Element, sdf_root_dir: str) -> List[str]:
+    """
+    Print xsd for top level SDF element
+    """
+    lines = []
+
+    elem_name = get_attribute(element, "name")
+    elem_type = get_attribute(element, "type")
+
+    elements = element.findall("element")
+    attributes = element.findall("attribute")
+    includes = element.findall("include")
+
+    lines.extend(print_documentation(element))
+    lines.append(
+        "<xsd:include schemaLocation='http://sdformat.org/schemas/types.xsd'/>"
+    )
+
+    # Reference any includes in the SDF file
+    for include in includes:
+        lines.extend(print_include(include))
+
+    if len(elements) or len(attributes) or len(includes):
+        lines.append(f"<xsd:element name='{elem_name}'>")
+        lines.append("  <xsd:complexType>")
+
+        if elem_name != "plugin" and (len(elements) or len(includes)):
+            lines.append("    <xsd:choice maxOccurs='unbounded'>")
+
+        for child_element in elements:
+            if "copy_data" in child_element.attrib:
+                element_lines = print_plugin_element(child_element)
+                lines.extend(indent_lines(element_lines, 4))
+            else:
+                element_lines = print_element(child_element)
+                lines.extend(indent_lines(element_lines, 6))
+
+        for include_element in includes:
+            element_lines = print_include_ref(include_element, sdf_root_dir)
+            lines.extend(indent_lines(element_lines, 6))
+
+        if elem_name != "plugin" and (len(elements) or len(includes)):
+            lines.append("    </xsd:choice>")
+
+        for attribute_element in attributes:
+            lines.extend(indent_lines(print_attribute(attribute_element), 4))
+
+        lines.append("  </xsd:complexType>")
+        lines.append("</xsd:element>")
+    else:
+        if elem_type and is_std_type(elem_type):
+            elem_type = xsd_type_string(elem_type)
```

Comment:
From https://github.com/gazebosim/sdformat/blob/bfeaf0738be02742434092b97a55a1f32b5f2e2f/tools/xmlschema.rb#L232-L243

```suggestion
            elem_type = f' type={xsd_type_string(elem_type)}'
```